library/src/scala/runtime/DynamicTuple.scala

```diff
@@ -264,6 +264,68 @@ object DynamicTuple {
         .asInstanceOf[Map[This, F]]
   }
 
+  def dynamicTake[This <: Tuple, N <: Int](self: This, n: N): Take[This, N] = {
+    if (n < 0) throw new IndexOutOfBoundsException(n.toString)
+    val actualN = Math.min(n, self.size)
+
+    type Result = Take[This, N]
+    val arr = (self: Any) match {
+      case () => Array.empty[Object]
+      case xxl: TupleXXL =>
+        xxl.elems.asInstanceOf[Array[Object]].take(actualN)
+      case _ =>
+        val arr = new Array[Object](actualN)
+        self.asInstanceOf[Product].productIterator.asInstanceOf[Iterator[Object]]
+          .copyToArray(arr, 0, actualN)
+        arr
+    }
+    dynamicFromIArray(arr.asInstanceOf[IArray[Object]]).asInstanceOf[Result]
+  }
+
+  def dynamicDrop[This <: Tuple, N <: Int](self: This, n: N): Drop[This, N] = {
+    if (n < 0) throw new IndexOutOfBoundsException(n.toString)
+    val size = self.size
+    val actualN = Math.min(n, size)
+
+    type Result = Drop[This, N]
+    val arr = (self: Any) match {
+      case () => Array.empty[Object]
```

Comment:
I'd suggest that most of these operations are unlikely to be called on Unit, so perhaps it adds unnecessary overhead to check for it first

Reply:
We need to handle it

Reply:
We need to handle it

Reply:
If unit was a subclass of Product, it would ease everything. But since it is not the case, I don't think there really is a better way of doing it...

Reply:
But we could remove the `case ()` and move the logic inside `case _ =>`